tests/queries/0_stateless/01550_create_map_type.sql

```diff
@@ -0,0 +1,40 @@
+-- String type
+drop table if exists table_map;
+create table table_map (a Map(String, String)) engine = Memory;
+insert into table_map values ({'name':'zhangsan', 'gender':'male'}), ({'name':'lisi', 'gender':'female'});
+select a['name'] from table_map;
+drop table if exists table_map;
+
+
+drop table if exists table_map;
+create table table_map (a Map(String, UInt64)) engine = MergeTree() order by a;
+insert into table_map select map('key1', number, 'key2', number * 2) from numbers(1111, 3);
+select a['key1'], a['key2'] from table_map;
+drop table if exists table_map;
+
+-- MergeTree Engine
+drop table if exists table_map;
+create table table_map (a Map(String, String), b String) engine = MergeTree() order by a;
+insert into table_map values ({'name':'zhangsan', 'gender':'male'}, 'name'), ({'name':'lisi', 'gender':'female'}, 'gender');
+select a[b] from table_map;
+select b from table_map where a = map('name','lisi', 'gender', 'female');
```

Comment:
Do we have a map constructor that accepts parallel arrays of keys/values a-la `sumMap`? And functions like `keys`/`values`/`items` that deconstruct it back into parallel arrays or an array of tuples. This constructor looks like it only works for a statically known set of keys, which might be not very practical.

Reply:
Map can be crated from tuple of arrays by using `CAST`. Like this: 
```sql
CAST(([1, 2, 3], ['1', '2', 'foo']), 'Map(UInt8, String)')
````. Maybe we need more convenient syntax to do that.

`keys` and `items` columns will be available after #17310 will be merged. `items` can be implemented also as a `CAST`.